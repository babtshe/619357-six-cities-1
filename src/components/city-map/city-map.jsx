import React, {PureComponent} from 'react';
import {propTypes} from './city-map.props';
import leaflet from 'leaflet';

class CityMap extends PureComponent {
  constructor(props) {
    super(props);
    this.containerRef = React.createRef();
  }

  componentDidMount() {
    if (this.props.locations.length) {
      const city = [52.38333, 4.9];
      const icon = leaflet.icon({
        iconUrl: `img/icon-marker.svg`,
        iconSize: [27, 39],
      });
      const zoom = 12;
      const map = leaflet.map(this.containerRef.current, {
        center: city,
        zoom,
        zoomControl: false,
        marker: true
      });
      map.setView(city, zoom);

      leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);

      this.props.locations.forEach((location) => {
        leaflet
        .marker(location, {icon})
        .addTo(map);
      });
    }
  }

  render() {
    return <div id="map" style={{
      width: `100%`,
      minHeight: `800px`,
      height: `100%`,
    }} ref={this.containerRef}></div>;
  }
}

CityMap.propTypes = propTypes;

export {CityMap};
