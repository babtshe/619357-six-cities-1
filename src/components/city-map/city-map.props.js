import {PropTypes} from 'prop-types';
import {coordinates} from '../../custom-prop-types/coordinates.prop';

export const propTypes = {
  locations: PropTypes.arrayOf(coordinates).isRequired,
  mapProvider: PropTypes.object,
};
