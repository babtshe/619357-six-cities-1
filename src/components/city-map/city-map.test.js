import React from 'react';
import renderer from 'react-test-renderer';
import {CityMap} from './city-map';

jest.mock(`leaflet`);

it(`CityMap should render correctly`, ()=>{
  const mockLocations = [];
  const tree = renderer
  .create(
      <CityMap
        locations={mockLocations}
      />)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
