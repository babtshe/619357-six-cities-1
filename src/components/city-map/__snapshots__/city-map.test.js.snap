// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityMap should render correctly 1`] = `
<div
  id="map"
  style={
    Object {
      "height": "100%",
      "minHeight": "800px",
      "width": "100%",
    }
  }
/>
`;
